packages/edit-site/src/components/sidebar/spacing-panel.js

```diff
@@ -65,29 +72,54 @@ function useCustomUnits( { units, contextName } ) {
 	return usedUnits.length === 0 ? false : usedUnits;
 }
 
-export default function SpacingPanel( {
-	context: { name },
-	getStyle,
-	setStyle,
-} ) {
+function useCustomSides( blockName, feature ) {
+	const support = getBlockSupport( blockName, 'spacing' );
+
+	// Skip when setting is boolean as theme isn't setting arbitrary sides.
+	if ( typeof support[ feature ] === 'boolean' ) {
+		return;
+	}
+
+	return support[ feature ];
+}
+
+function filterValuesBySides( values, sides ) {
+	if ( ! sides ) {
+		// If no custom side configuration all sides are opted into by default.
+		return values;
+	}
+
+	// Only include sides opted into within filtered values.
+	const filteredValues = {};
+	sides.forEach( ( side ) => ( filteredValues[ side ] = values[ side ] ) );
+
+	return filteredValues;
+}
+
+export default function SpacingPanel( { context, getStyle, setStyle } ) {
+	const { name } = context;
+	const showPaddingControl = useHasPadding( context );
 	const units = useCustomUnits( { contextName: name, units: CSS_UNITS } );
+
 	const paddingValues = getStyle( name, 'padding' );
-	const setPaddingValues = ( { top, right, bottom, left } ) => {
-		setStyle( name, 'padding', {
-			top: top || paddingValues?.top,
-			right: right || paddingValues?.right,
-			bottom: bottom || paddingValues?.bottom,
-			left: left || paddingValues?.left,
-		} );
+	const paddingSides = useCustomSides( name, 'padding' );
+
+	const setPaddingValues = ( newPaddingValues ) => {
+		const padding = filterValuesBySides( newPaddingValues, paddingSides );
+		setStyle( name, 'padding', padding );
 	};
+
 	return (
 		<PanelBody title={ __( 'Spacing' ) }>
-			<BoxControl
-				values={ paddingValues }
-				onChange={ setPaddingValues }
-				label={ __( 'Padding' ) }
-				units={ units }
-			/>
+			{ showPaddingControl && (
```

Comment:
Should we  hide the panel entirely if there's no padding control?

Reply:
Unless I've missed something, we already do.

Within `global-styles-sidebar.js` there is a check using `useHasSpacingPanel` as to whether the `SpacingPanel` should be displayed. This matches the current approach for the typography, color and border panels.

The `showPaddingControl` check here within the component is because we'll soon be [adding a control for margins](https://github.com/WordPress/gutenberg/pull/30608/files?file-filters%5B%5D=.js#diff-bb729c1a11a1203794da39c490560c4c9ac0031906d9981794b9d32d372f1d63R138-R146) as well.

Reply:
I guess I was just wondering why this component (taken separately) renders an empty panel if padding is not shown (I'm sure this will change with the addition of margins though). This is not that important.